spec/tree_spec.rb

```diff
@@ -1,14 +1,113 @@
-require 'rspec'
-require 'tree'
+require "rspec"
+require "tree"
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe Tree do
+  let(:tree) { Tree.new }
+
+  it "should be a Class" do
+    expect(described_class.is_a? Class).to be true
   end
-end
 
-describe 'Fruit' do
+  context "when tree is starting" do
+    it "should not have age" do
+      expect(tree.age).to eq(0)
+    end
+
+    it "should not have height" do
+      expect(tree.age).to eq(0)
+    end
+
+    it "should not have apples" do
+      expect(tree.any_apples?).to be false
+    end
+
+    it "should be alive" do
+      expect(tree.dead?).to be false
+    end
+  end
+
+  context "when tree is maturing" do
+    it "should age" do
+      starting_age = tree.age
+      tree.age!
+      expect(tree.age - starting_age).to be > 0
+    end
+
+    it "should increase height" do
+      starting_height = tree.height
+      tree.age!
+      expect(tree.height - starting_height).to be > 0
+    end
+
+    it "should stop increasing height after 5 years" do
+      5.times { tree.age! }
+      height_after_5_years = tree.height
+      tree.age!
+      expect(tree.height).to eq(height_after_5_years)
+    end
+
+    it "should produce apples after 5 years" do
+      4.times { tree.age! }
+      expect(tree.any_apples?).to be false
+      tree.age!
+      expect(tree.any_apples?).to be true
```

Comment:
http://www.relishapp.com/rspec/rspec-expectations/v/3-1/docs/built-in-matchers/change-matcher
`expect { an_object.modify }.to change { an_object }.from(some_state).to(another_state)`